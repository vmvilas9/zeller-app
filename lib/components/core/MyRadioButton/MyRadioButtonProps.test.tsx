import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {MyRadioButton} from './index';

describe('MyRadioButton', () => {
    it('renders correctly when unselected', () => {
        const tree = renderer
            .create(<MyRadioButton text="Test Radio" selected={false} onSelect={() => {
            }}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when selected', () => {
        const tree = renderer
            .create(<MyRadioButton text="Test Radio" selected={true} onSelect={() => {
            }}/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('displays the correct text', () => {
        const {getByText} = render(
            <MyRadioButton text="Test Radio Button" selected={false} onSelect={() => {
            }}/>
        );

        expect(getByText('Test Radio Button')).toBeTruthy();
    });

    it('applies the correct style when unselected', () => {
        const {getByTestId, queryByTestId} = render(
            <MyRadioButton text="Test Radio" selected={false} onSelect={() => {
            }}/>
        );

        const outerCircle = getByTestId('radio-outer-circle');
        expect(outerCircle.props.style).toEqual(
            expect.objectContaining([
                {
                    "backgroundColor": "#FFFFFF"
                },
                {
                    "alignItems": "center",
                    "borderColor": "#0171CE",
                    "borderRadius": 12,
                    "borderWidth": 2,
                    "height": 24,
                    "justifyContent": "center",
                    "marginRight": 12,
                    "width": 24
                }
            ]),
        );

        expect(queryByTestId('radio-inner-circle')).toBeNull()
    });

    it('applies the correct style when selected', () => {
        const {getByTestId} = render(
            <MyRadioButton text="Test Radio" selected={true} onSelect={() => {
            }}/>
        );

        const innerCircle = getByTestId('radio-inner-circle');
        expect(innerCircle).toBeTruthy();
    });

    it('calls onSelect when pressed', () => {
        const mockOnSelect = jest.fn();

        const {getByText} = render(
            <MyRadioButton text="Test Radio" selected={false} onSelect={mockOnSelect}/>
        );

        fireEvent.press(getByText('Test Radio'));
        expect(mockOnSelect).toHaveBeenCalledTimes(1);
    });
});
