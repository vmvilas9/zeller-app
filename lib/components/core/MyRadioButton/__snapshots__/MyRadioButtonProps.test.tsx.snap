// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRadioButton renders correctly when selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#E8F2FB",
      "borderRadius": 8,
      "flexDirection": "row",
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
        },
        {
          "alignItems": "center",
          "borderColor": "#0171CE",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        },
      ]
    }
    testID="radio-outer-circle"
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "backgroundColor": "#0171CE",
            "borderRadius": 6,
            "height": 12,
            "width": 12,
          },
        ]
      }
      testID="radio-inner-circle"
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Test Radio
  </Text>
</View>
`;

exports[`MyRadioButton renders correctly when unselected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 8,
      "flexDirection": "row",
      "marginTop": 10,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
        },
        {
          "alignItems": "center",
          "borderColor": "#0171CE",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        },
      ]
    }
    testID="radio-outer-circle"
  />
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Test Radio
  </Text>
</View>
`;
